library/javatests/net/openid/appauth/ClientSecretBasicTest.java

```diff
@@ -41,6 +41,21 @@ public void testGetRequestHeaders() {
         assertThat(headers).containsEntry("Authorization", expectedAuthzHeader);
     }
 
+    @Test
+    public void getGetRequestHeaders_idAndSecretAreUrlEncoded() {
+        String secretThatChangesWhenEncoded = "1/2_3+4";
```

Comment:
How is space going to be encoded?

Reply:
Sure, I'll add those. The flippant answer to your question is "however java.net.URLEncoder decides to handle it" 😆